lib/connector-handler.ts

```diff
@@ -49,8 +49,30 @@ export interface Context {
 	[prop: string]: any
 
 	reloadConfig(): Promise<any>
+	assumeAwsRole(externalRoleInput: ExternalRoleInput): Promise<CredentialResponse>;
+}
+export class ExternalRoleInput {
+	externalArnRole: string;
+	externalId     : string;
+	roleSessionName :string;
+	constructor(externalArnRole: string, externalId: string, roleSessionName: string) {
+		this.externalArnRole = externalArnRole;
+		this.externalId = externalId;
+		this.roleSessionName = roleSessionName;
+	}
+}
+export class CredentialResponse {
+    accessKeyId: string;
+    secretAccessKey: string;
+    sessionToken: string;
+	expiration : number;
+    constructor(accessKeyId: string, secretAccessKey: string, sessionToken: string, expiration: number) {
+		this.accessKeyId = accessKeyId;
+		this.secretAccessKey = secretAccessKey;
+		this.sessionToken = sessionToken;
+		this.expiration = expiration;
+	}
```

Comment:
Could we call these something like `AssumeAwsRoleRequest` and `AssumeAwsRoleResponse`? Just to line up all the names.